tests/array_view_test.py

```diff
@@ -65,9 +65,9 @@ def test_set_index(self):
         uind = ak.cast(iind, ak.uint64)
 
         inav[nind] = -9999
-        unav[nind] = -9999
+        unav[nind] = 2**64 - 9999
         iav[uind] = -9999
-        uav[iind] = -9999
+        uav[iind] = 2**64 - 9999
```

Comment:
I want to discuss this a bit more, specifically for setting a value in a `pdarray`. I think overall this change is going to have larger implications. Especially, because a user will still be able to call `uav[iind] = -9999`.

I think we need to document this at a minimum. Especially, seen as these updates do not address the areas where the warning can be produced within the main code, they are updating the format to not produce the warning. However, it can still happen within our code. Not sure if there is a good way to address, but think it warrants some more discussion.

Reply:
Thanks for the feedback! I updated `format_other` to allow us to keep using vals that are out of bounds without having the numpy deprecation warning
```python
>>> ak.array([2**64 - 1, 0, -1], dtype=ak.uint64)
array([18446744073709551615 0 18446744073709551615])

>>> pda = ak.arange(10, dtype=ak.uint64)
>>> pda[5:10] = -2
>>> pda
array([0 1 2 3 4 18446744073709551614 18446744073709551614 18446744073709551614 18446744073709551614 18446744073709551614])
```